core/src/main/java/com/rudderstack/android/sdk/core/RudderClient.java

```diff
@@ -284,17 +288,26 @@ public void screen(@NonNull String screenName, @Nullable RudderProperty property
      * Identify your user
      *
      * @param message instance of RudderMessage
+     * @deprecated Will be removed soon
      */
     public void identify(@NonNull RudderMessage message) {
         // update cached traits and persist
         RudderElementCache.updateTraits(message.getTraits());
         RudderElementCache.persistTraits();
 
+        //  handle external Ids
+        RudderOption option = message.getRudderOption();
+        if (option != null && option.getExternalIds() != null && !option.getExternalIds().isEmpty()) {
```

Comment:
Can we store `option.getExternalIds()` in a variable instead of calling it again?